cmd/operator-sdk/generate/packagemanifests/cmd.go

```diff
@@ -25,110 +26,68 @@ import (
 	operatorName string
 	version      string
 	fromVersion  string
 	inputDir     string
 	outputDir    string
+	kustomizeDir string
 	deployDir    string
-	apisDir      string
 	crdsDir      string
 	updateCRDs   bool
 	stdout       bool
 	quiet        bool
 
-	// Interactive options.
-	interactiveLevel projutil.InteractiveLevel
-	interactive      bool
-
 	// Package manifest options.
 	channelName      string
 	isDefaultChannel bool
 }
 
+//nolint:maligned
+type packagemanifestsCmdLegacy struct {
+	packagemanifestsCmd
+
+	apisDir          string
+	interactiveLevel projutil.InteractiveLevel
+	interactive      bool
+}
+
 // NewCmd returns the 'packagemanifests' command configured for the new project layout.
 func NewCmd() *cobra.Command {
 	c := &packagemanifestsCmd{}
 
 	cmd := &cobra.Command{
 		Use:     "packagemanifests",
-		Short:   "Generates a package manifests format",
+		Short:   "Generates package manifests data for the operator",
 		Long:    longHelp,
 		Example: examples,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			if len(args) != 0 {
 				return fmt.Errorf("command %s doesn't accept any arguments", cmd.CommandPath())
 			}
 
-			// Check if the user has any specific preference to enable/disable
-			// interactive prompts. Default behaviour is to disable the prompt
-			// unless a base package does not exist.
-			if cmd.Flags().Changed("interactive") {
-				if c.interactive {
-					c.interactiveLevel = projutil.InteractiveOnAll
-				} else {
-					c.interactiveLevel = projutil.InteractiveHardOff
-				}
-			}
-
-			// Generate kustomize bases and manifests by default if no flags are set
-			// so the default behavior is "do everything".
-			fs := cmd.Flags()
-			if !fs.Changed("kustomize") && !fs.Changed("manifests") {
-				c.kustomize = true
-				c.manifests = true
-			}
-
 			cfg, err := kbutil.ReadConfig()
 			if err != nil {
 				log.Fatal(fmt.Errorf("error reading configuration: %v", err))
 			}
-			c.setCommonDefaults(cfg)
+			c.setDefaults(cfg)
 
-			if c.kustomize {
-				if err = c.runKustomize(cfg); err != nil {
-					log.Fatalf("Error generating package bases: %v", err)
-				}
+			if err = c.validate(); err != nil {
+				return fmt.Errorf("invalid command options: %v", err)
 			}
-			if c.manifests {
-				if err = c.validateManifests(); err != nil {
-					return fmt.Errorf("invalid command options: %v", err)
-				}
-				if err = c.runManifests(cfg); err != nil {
-					log.Fatalf("Error generating package manifests: %v", err)
-				}
+			if err = c.run(cfg); err != nil {
+				log.Fatalf("Error generating package manifests: %v", err)
 			}
 
 			return nil
 		},
 	}
 
-	cmd.Flags().BoolVar(&c.kustomize, "kustomize", false, "Generate kustomize bases")
-	cmd.Flags().BoolVar(&c.manifests, "manifests", false, "Generate package manifests")
+	cmd.Flags().StringVar(&c.kustomizeDir, "kustomize-dir", filepath.Join("config", "manifests"),
+		"Directory containing kustomize bases and a kustomization.yaml")
```

Comment:
Nit: mention what the Kustomize directory is for if users are allowed to configure it.
```suggestion
		"Directory containing Kustomize bases and a kustomization.yaml for generating CSV manifests")
```

Reply:
Good catch, although kustomize isn't capitalized on their GH repo docs.